modules/protocol/src/vector.ts

```diff
@@ -34,6 +40,17 @@ export class Vector implements IVectorProtocol {
     [ProtocolEventName.CHANNEL_UPDATE_EVENT]: Evt.create<ChannelUpdateEvent>(),
   };
 
+  // This holds the highest seen nonce (proposed or received) for each channel.
+  // Will be used to determine the priority ordering of the queue currently
+  // being executed
+  private highestNonce: { [channelAddr: string]: number } = {};
```

Comment:
TypeScript has a dedicated `Map<Key, Value>` for this sort of thing.

Reply:
Are you talking about using a `Map` class (with the `get`, `set`, `has`) interface or using just the `Map<K, V>` type? If it's just the type I personally prefer the typing above since then you get more context for what the `K` type should be instead of just `string`

Reply:
Ah. So you like that it is self documenting there. Another way to approach that is have `K, V` use tagged types. Eg: `Map<ChannelAddr, Nonce>`. The problem with objects generally is that you can shadow properties, and there are performance implications. Those are probably not a huge deal here (the chances that an address will be the name of a property is 0), but sometimes it's a security risk.

Reply:
There's also `Record<Key, Value>`, which may be more appropriate.